src/Tpay/Provider/TpayApiChannelListProvider.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Tpay\Provider;
+
+use CommerceWeavers\SyliusTpayPlugin\Payum\Exception\UnableToGetBankListException;
+use CommerceWeavers\SyliusTpayPlugin\Tpay\Resolver\TpayTransactionChannelResolverInterface;
+
+final class TpayApiChannelListProvider implements TpayApiChannelListProviderInterface
+{
+    public function __construct(
+        private readonly TpayTransactionChannelResolverInterface $channelResolver,
+    ) {
+    }
+
+    public function provide(): array
+    {
+        $result = $this->channelResolver->resolve();
+
+        if (!isset($result['result']) || 'success' !== $result['result']) {
+            throw new UnableToGetBankListException('Unable to get channels list. Response: ' . json_encode($result));
+        }
+
+        return $result['channels'];
+    }
+}
```

Comment:
I have mixed feelings about this class, we can put these logic directly in the resolver. wdyt @coldic3 @lchrusciel

Reply:
I have nothing against this provider itself, except it's not unit tested :grinning:

It's used only in the validator, so if you decide to move it into the private method, I'm ok with that. If not, this should be tested with PHPUnit.